### PR TITLE
ci: add windows test

### PR DESCRIPTION
**Pull request by SoonIter**
## Summary

1. add windows test yml
2. add path-serializer as vitest snapshot serializer

### CODE REVIEW

**Comment by Timeless0911** [comment]
We need path serializer.

**Comment by fi3ework** [comment]
@SoonIter If you're not in a hurry, I'll resolve https://github.com/web-infra-dev/rsbuild/issues/3099 by tomorrow night at the latest. Or you can claim this task.

**Comment by SoonIter** [comment]
> @SoonIter If you're not in a hurry, I'll resolve https://github.com/web-infra-dev/rsbuild/issues/3099 by tomorrow night at the latest. Or you can claim this task if you have enough spare time.

you can name the package repo and assign it to me

**Comment by SoonIter** [comment]
> What about [npmjs.com/package/serializer-path](https://www.npmjs.com/package/serializer-path) or [npmjs.com/package/path-serializer](https://www.npmjs.com/package/path-serializer), not claimed and easy to search without any prefix/suffix.

[path-serializer](https://www.npmjs.com/package/path-serializer) is better

**Comment by SoonIter** [comment]
done

File: packages/core/tests/lcp.test.ts [hunk 1]
```diff
@@ -19,35 +20,57 @@ describe('LCP calculate correctly', () => {
   it('correct 1', async () => {
     vol.fromJSON({ '/Users/Someone/project-a/src': null });
 
-    const result = await calcLongestCommonPath([
-      '/Users/Someone/project-a/src/helpers',
-      '/Users/Someone/project-a/src',
-      '/Users/Someone/project-a/src/utils',
-    ]);
-    expect(result).toBe('/Users/Someone/project-a/src');
+    if (path.sep === '/') {
```
Comment (fi3ework) [inline packages/core/tests/lcp.test.ts:23]:
```suggestion
    if (os.platform() !== 'win32') {
```

Could make the code more clear.

**Comment by SoonIter** [comment]
done

**Comment by fi3ework** [comment]
Good job! The `path-serializer` has become the first published package built by Rslib (with help of manual magic).

**Review by fi3ework — APPROVED** [review]

**Pull request merged**